### ISSUE TITLE
Request for capability to delete specific entries from secure storage

### ISSUE THREAD

**Issue by vanderzee-anl-gov** [eclipse-packaging/packages#215]
Under the Preferences window in General -> Security -> Secure Storage, one can view the contents of secure storage and can see, for example, that there may be different entries in the Default Secure Storage for org.eclipse.core.net.proxy.auth, org.sonarlint.eclipse.core, GIT, and org.eclipse.equinox.secure.storage

I would like the capability to delete (for the purposes of updating) the secure storage associated with GIT without deleting any of the other entries in Secure Storage. There is a "Delete" button that is currently provided, but that button deletes the entire secure storage. At present, if I store my Git token in Eclipse secure storage, then when my Git token changes I am forced to delete the entire secure storage. When the token changes, git authentication simply fails and does not prompt me for a new token unless I delete the Eclipse secure storage, since that is the only way that I can find to delete the expired token.

**Comment by merks** [comment]
Note that if you install this:

![image](https://github.com/user-attachments/assets/220c053a-5ade-42da-aec5-fb3b19135910)

Then the following will be available in your Preferences:

![image](https://github.com/user-attachments/assets/15c00f5a-45da-4ae7-a891-89e8cbb05278)

It has an Edit... that lets you open in an editor:

![image](https://github.com/user-attachments/assets/f20fdad8-8afd-416c-829d-b546216c6e70)

The value displayed for the password is not directly the clear text value but rather is encrypted by a generated UUID.

You can delete nodes in this editor and save them, which saves them to the actual underlying preferences, including the secure preference store.

If you double click and node, you can see the values in the Properties view where you can edit them.  Again, the passwords are encrypted there too.  But if you change the filter (funnel icon on the toolbar) to show advanced properties, you can change the `secure` property to false, and then you can finally see your clear-text password and modify it as needed.

![image](https://github.com/user-attachments/assets/bea1a105-194b-4817-9951-0afdd30f6238)

**Comment by merks** [comment]
The preference dialog you mention is implemented by 

https://github.com/eclipse-equinox/equinox/blob/master/bundles/org.eclipse.equinox.security.ui/src/org/eclipse/equinox/internal/security/ui/storage/StoragePreferencePage.java

So if you want that dialog to provide direct support for what you describe, which seems reasonable, you'll need to open an issue there:

https://github.com/eclipse-equinox/equinox/issues

Unfortunately I cannot transfer you issue to a different organization so I will need to close it here.

You might consider contributing your desired functionality:

https://github.com/eclipse-equinox/.github/blob/main/CONTRIBUTING.md

**Comment by vanderzee-anl-gov** [comment]
Thanks. Once I get these preferences open in the editor, I can also delete just the GIT node, which does not delete the rest of the secure storage but is enough to get Git to ask me for a new token. This avoids storing the token in plain text in case I don't want to do that.

**Comment by merks** [comment]
Yes, you can just delete a single node and save and then be prompted again.  I expect you can just delete the password node and be prompted, but I haven't tried that.  Note that editing a password will not save it as plain text but rather will save it exactly the same way as it is saved when you are prompted by a password dialog so no security worries there...

**Comment by vanderzee-anl-gov** [comment]
I did try deleting just the password node, and that did not work for me.

**Comment by merks** [comment]
I see.  The underlying storage might return a null or empty password in that case, and then you'd still not be prompted that the password is wrong.  Good to know...
